packages/idea-transformer/src/Transformer.ts

```diff
@@ -49,16 +49,19 @@ export default class Transformer<T extends Record<string, unknown>> {
             //make sure there is a schema type
             schema.type = schema.type || {};
             //loop through child types
-            for (const [ name, type ] of Object.entries(child.type)) {
+            for (const [name, type] of Object.entries(child.type)) {
               const parent = schema.type[name];
               //if type from child doesn't exist in schema (parent)
               if (!parent) {
                 //add it to schema (parent)
                 schema.type[name] = type;
                 continue;
-              //if parent isnt final
-              } else if (parent.mutable) {
-                //soft merge type into parent
+                //if parent isnt final
+              }
+              //I remove the else to make the code easier to read and
+              //understand and to check separete the condition
```

Comment:
Remove commit and code change notes. If someone were to randomly read this file, this would have no context. We should use notes to explain the current logic.